### ISSUE TITLE
False-positive `empty-body` when inheriting from protocol that inherits from `Protocol`

### ISSUE THREAD

**Issue by bersbersbers** [python/mypy#17163]
**Bug Report**
`mypy` accepts that `BaseProtocol.foo` does not have a body, but want `DerivedProtocol.bar` to have a body. `pyright` does not have this problem.

**To Reproduce**

```python
from typing import Protocol

class BaseProtocol(Protocol):
    def foo(self) -> int: ...

class DerivedProtocol(BaseProtocol):
    def bar(self) -> int: ...
```

https://mypy-play.net/?mypy=latest&python=3.12&gist=085d36ec959e50bab990a37fdacdf3b9

**Expected Behavior**

No error

**Actual Behavior**

```
main.py:7: error: Missing return statement  [empty-body]
main.py:7: note: If the method is meant to be abstract, use @abc.abstractmethod
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.9.0
- Mypy command-line flags: nothing
- Mypy configuration options from `mypy.ini` (and other config files): nothing
- Python version used: 3.12.1

**Comment by JelleZijlstra** [comment]
This error is correct. `DerivedProtocol` is a concrete class, not a protocol, because it does not inherit directly from `Protocol`.

**Comment by bersbersbers** [comment]
You are right of course. Thank you!
